js_modules/dagster-ui/packages/ui-core/src/nav/useCodeLocationsStatus.tsx

```diff
@@ -1,31 +1,30 @@
-import {gql, useQuery} from '@apollo/client';
 import {Box, ButtonLink, Colors} from '@dagster-io/ui-components';
-import {useCallback, useContext, useState} from 'react';
+import {useCallback, useContext, useEffect, useLayoutEffect, useState} from 'react';
 import {useHistory} from 'react-router-dom';
+import {atom, useRecoilValue} from 'recoil';
 import styled from 'styled-components';
 
-import {
-  CodeLocationStatusQuery,
-  CodeLocationStatusQueryVariables,
-} from './types/useCodeLocationsStatus.types';
 import {showSharedToaster} from '../app/DomUtils';
-import {useQueryRefreshAtInterval} from '../app/QueryRefresh';
 import {RepositoryLocationLoadStatus} from '../graphql/types';
 import {StatusAndMessage} from '../instance/DeploymentStatusType';
 import {WorkspaceContext} from '../workspace/WorkspaceContext';
+import {CodeLocationStatusQuery} from '../workspace/types/WorkspaceQueries.types';
 
 type LocationStatusEntry = {
   loadStatus: RepositoryLocationLoadStatus;
   id: string;
   name: string;
 };
 
-const POLL_INTERVAL = 5 * 1000;
-
 type EntriesById = Record<string, LocationStatusEntry>;
 
-export const useCodeLocationsStatus = (skip = false): StatusAndMessage | null => {
-  const {locationEntries, refetch} = useContext(WorkspaceContext);
+export const codeLocationStatusAtom = atom<CodeLocationStatusQuery | undefined>({
```

Comment:
This exists for perf. I don't want to put this in WorkspaceContext since it's polled every few seconds and would end up causing a lot of components to re-render (most of which aren't using React.memo)